yb-voyager/src/srcdb/postgres.go

```diff
@@ -1434,43 +1460,61 @@ func (pg *PostgreSQL) listSequencesMissingSelectPermission() (sequencesWithMissi
 	return sequencesWithMissingPerm, nil
 }
 
-func (pg *PostgreSQL) listTablesMissingSelectPermission() (tablesWithMissingPerm []string, err error) {
+func (pg *PostgreSQL) listTablesMissingSelectPermission(queryTableList string) (tablesWithMissingPerm []string, err error) {
 	// Users only need SELECT permissions on the tables of the schema they want to export for export schema
-	trimmedSchemaList := pg.getTrimmedSchemaList()
-	trimmedSchemaList = append(trimmedSchemaList, "pg_catalog", "information_schema")
-	querySchemaList := "'" + strings.Join(trimmedSchemaList, "','") + "'"
+	checkTableSelectPermissionQuery := ""
+	if queryTableList == "" {
```

Comment:
this is essentially for the `export-schema` and `assess-migration` case where we don't have table list so need to check on all tables in schema right?

Reply:
Yes correct